File: frontend-flutter/lib/services/auth_service.dart
import '../config/api_config.dart';
import 'api_service.dart';

class AuthService {
  final _api = ApiService();

  Future<String> login(String username, String password) async {
    final response = await _api.dio.post(
      ApiConfig.loginEndpoint,
      data: {'username': username, 'password': password},
    );
    final token = response.data['id_token'];
    await _api.saveToken(token);
    return token;
  }

  Future<void> register(String login, String email, String password, String? firstName, String? lastName) async {
    await _api.dio.post(
      ApiConfig.registerEndpoint,
      data: {
        'login': login,
        'email': email,
        'password': password,
        'firstName': firstName,
        'lastName': lastName,
      },
    );
  }

  Future<Map<String, dynamic>> getAccount() async {
    final response = await _api.dio.get(ApiConfig.accountEndpoint);
    return response.data;
  }

  Future<void> logout() async {
    await _api.deleteToken();
  }

  Future<bool> isLoggedIn() async {
    final token = await _api.getToken();
    return token != null;
  }
}

File: frontend-react/src/services/authService.ts
import api, { ENDPOINTS } from '../config/api';

export interface LoginRequest {
  username: string;
  password: string;
}

export interface LoginResponse {
  id_token: string;
}

export interface RegisterRequest {
  login: string;
  email: string;
  password: string;
  firstName?: string;
  lastName?: string;
}

export const authService = {
  login: async (data: LoginRequest): Promise<LoginResponse> => {
    const response = await api.post(ENDPOINTS.AUTH.LOGIN, data);
    return response.data;
  },

  register: async (data: RegisterRequest): Promise<void> => {
    await api.post(ENDPOINTS.AUTH.REGISTER, data);
  },

  getAccount: async () => {
    const response = await api.get(ENDPOINTS.AUTH.ACCOUNT);
    return response.data;
  },
};
